precompiles/assets-erc20/src/lib.rs

```diff
@@ -124,30 +126,58 @@ where
 			// this makes the precompiles dependent on such a pallet, which is not ideal
 			if !pallet_assets::Pallet::<Runtime, Instance>::total_supply(asset_id).is_zero() {
```

Comment:
We can probably use here Self:is_precompile

Reply:
But that would require to compute `asset_id` 2 times (in is_precompile and in this function since you need this data). But if it has an insignificant cost indeed that would be cleaner.

Reply:
sorry I meant using Self::is_precompile for both checks (whether total supply is zeor, and whether assetId exists). I think you dont add new costs there

Reply:
Oh actually you were right, sorry, did not realize about the fact that we need the asset_id later on. Makes sense then!